### PR TITLE
refactor(cloud-storage): using new cloud storage api of the server v2

### PR DESCRIPTION
**Pull request by Cheerego7**
- refactor(flat-services): get whiteboard task data
- refactor(service-providers): change parameters of the queryConvertingTaskStatus function
- refactor(flat-stores): replace old server v1 api of the cloud storage and remove unused api
- refactor(flat-components): change the component props of the cloud storage

### CODE REVIEW

**Review by crimx — APPROVED** [review]

File: packages/flat-components/src/containers/CloudStorageContainer/CloudStorageContainer.stories.tsx [hunk 1]
```diff
@@ -163,19 +164,22 @@ class FakeStore extends CloudStorageStore {
...
-                    ]),
                     createAt: faker.date.past(),
                     fileURL: faker.internet.url(),
-                    taskToken: faker.random.word(),
-                    taskUUID: faker.random.word(),
-                    region: Region.CN_HZ,
-                    external: faker.datatype.boolean(),
                     resourceType: FileResourceType.NormalResources,
+                    meta: {
+                        whiteboardProjector: {
```
Comment (BlackHole1) [inline packages/flat-components/src/containers/CloudStorageContainer/CloudStorageContainer.stories.tsx:171]: You can add `whiteboardConvert`?

File: packages/flat-server-api/src/storage.ts [hunk 2]
```diff
@@ -76,84 +102,82 @@ interface UploadFinishPayload {
...
-}
-
-export interface RemoveExternalFilesPayload {
-    fileUUIDs: string[];
-}
-
-export async function removeExternalFiles(payload: RemoveExternalFilesPayload): Promise<void> {
-    if (payload.fileUUIDs.length > 0) {
-        await post("cloud-storage/url-cloud/remove", payload);
+    await postV2("cloud-storage/upload/cancel", payload || {});
```
Comment (BlackHole1) [inline packages/flat-server-api/src/storage.ts:156]: remove

File: packages/flat-server-api/src/storage.ts [hunk 3]
```diff
@@ -76,84 +102,82 @@ interface UploadFinishPayload {
...
+
+export function getWhiteboardTaskData(
+    resourceType: ResourceType,
+    meta: metaType,
+): { taskUUID: string; taskToken: string; convertStep: FileConvertStep } | null {
+    switch (resourceType) {
+        case "WhiteboardProjector": {
+            return {
+                taskUUID: meta.whiteboardProjector!.taskUUID!,
+                taskToken: meta.whiteboardProjector!.taskToken!,
```
Comment (BlackHole1) [inline packages/flat-server-api/src/storage.ts:167]: `taskUUID` and `taskToken` maybe is `undefined`

File: packages/flat-server-api/src/storage.ts [hunk 4]
```diff
@@ -76,84 +102,82 @@ interface UploadFinishPayload {
...
+                convertStep: meta.whiteboardProjector!.convertStep!,
+            };
+        }
+        case "WhiteboardConvert": {
+            return {
+                taskUUID: meta.whiteboardConvert!.taskUUID!,
+                taskToken: meta.whiteboardConvert!.taskToken!,
```
Comment (BlackHole1) [inline packages/flat-server-api/src/storage.ts:174]: `taskUUID` and `taskToken` maybe is `undefined`

File: service-providers/courseware/flat-courseware-netless/src/index.ts [hunk 5]
```diff
@@ -58,83 +63,86 @@ export class FlatCoursewareNetless extends FlatCourseware {
...
+                    try {
+                        await convertFinish({ fileUUID: file.fileUUID });
+                    } catch (e) {
+                        if (
+                            isServerRequestError(e) &&
+                            e.errorCode === RequestErrorCode.FileIsConverted
+                        ) {
+                            // ignore this error
+                            // there's another `convertFinish()` call in ./store.tsx
+                            // we call this api in two places to make sure the file is correctly converted (in server)
```
Comment (BlackHole1) [inline service-providers/courseware/flat-courseware-netless/src/index.ts:88]: remove

File: service-providers/file-preview-netless-slide/src/index.ts [hunk 6]
```diff
@@ -1,32 +1,35 @@
...
-            taskId: result.uuid,
-            url:
-                result.prefix ||
-                extractLegacySlideUrlPrefix(
-                    result.progress?.convertedFileList[0].conversionFileUrl,
-                ),
-        });
+        if (
+            file.resourceType === "WhiteboardConvert" ||
+            file.resourceType === "WhiteboardProjector"
```
Comment (BlackHole1) [inline service-providers/file-preview-netless-slide/src/index.ts:15]: use `enum`

**Pull request merged**